pyrate/merge.py

```diff
@@ -248,21 +245,43 @@ def assemble_tiles(s: Tuple, dir: str, tiles: Tile, out_type: str, index: Option
 
     # loop over each tile, load and slot in to correct spot
     for t in tiles:
-        tile_file = Path(join(dir, out_type + '_'+str(t.index)+'.npy'))
+        tile_file = Path(join(dir, out_type + '_' + str(t.index) + '.npy'))
         tile = np.load(file=tile_file)
-        if index is None:  #2D array
+        if index is None:  # 2D array
             merged_array[t.top_left_y:t.bottom_right_y, t.top_left_x:t.bottom_right_x] = tile
-        else:  #3D array
+        else:  # 3D array
             merged_array[t.top_left_y:t.bottom_right_y, t.top_left_x:t.bottom_right_x] = tile[:, :, index]
 
     log.debug('Finished assembling tiles for {}'.format(out_type))
     return merged_array
 
 
-def _save_merged_files(ifgs_dict, outdir, array, out_type, index=None, savenpy=None):
+out_type_md_dict = {
+    'stack_rate': ifc.STACKRATE,
+    'stack_error': ifc.STACKERROR,
+    'stack_samples': ifc.STACKSAMP,
+    'linear_rate': ifc.LINRATE,
+    'linear_error': ifc.LINERROR,
+    'linear_samples': ifc.LINSAMP,
+    'linear_intercept': ifc.LINICPT,
+    'linear_rsquared': ifc.LINRSQ,
+    'tsincr': ifc.INCR,
+    'tscuml': ifc.CUML
+}
+
+los_projection_out_types = {'tsincr', 'tscuml', 'linear_rate', 'stack_rate'}
+los_projection_divisors = {
+    ifc.LINE_OF_SIGHT: lambda data: 1,
+    ifc.PSEUDO_VERTICAL: lambda data: np.sin(data),
+    ifc.PSEUDO_HORIZONTAL: lambda data: np.cos(data)
```

Comment:
Following some thought on this (including hand drawn sketches!), I believe we have these wrong. The incidence angle is measured between the LOS vector and the nadir (vertical) direction at the ground location. So based on this we should have:
```
los_projection_multipliers = {
    ifc.LINE_OF_SIGHT: lambda data: 1,
    ifc.PSEUDO_VERTICAL: lambda data: np.cos(data),
    ifc.PSEUDO_HORIZONTAL: lambda data: np.sin(data)
```
The result of these multiplications with the input signal is a projected signal that is *smaller* than the LOS signal (LOS on the hypotenuse of right-angled triangle)

Reply:
After some discussion, I have realised that my suggestion is incorrect, and the implemented code is correct. So confusing! After projection to vertical, the signal will be *larger* than the LOS signal, because the vertical forms the hypotenuse side of the projection triangle.

Reply:
That's a good way to explain the scenario. Thanks @richardt94 !

Regarding "pseudo-horizontal", you are correct. The motion is in the horizontal plane (perpendicular to nadir). A future extension could be to actually estimate the pseudo-E-W or pseudo-N-S motion by bringing the azimuth angle in to the calculation. Since I don't think we will use this, I think it is low priority - maybe a suggestion for an external contributor on Github Issues page? https://github.com/GeoscienceAustralia/PyRate/issues